fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -3286,13 +3287,20 @@ public void createMaterializedView(CreateMaterializedViewStatement stmt)
                         .put(PropertyAnalyzer.PROPERTIES_STORAGE_COLDOWN_TIME,
                                 String.valueOf(dataProperty.getCooldownTimeMs()));
             }
-            if (properties != null && !properties.isEmpty()) {
+            if (properties.containsKey(PropertyAnalyzer.PROPERTIES_PARTITION_TTL_NUMBER)) {
+                int number = PropertyAnalyzer.analyzePartitionTimeToLive(properties);
+                materializedView.getTableProperty().getProperties()
+                        .put(PropertyAnalyzer.PROPERTIES_PARTITION_TTL_NUMBER, String.valueOf(number));
+                materializedView.getTableProperty().setPartitionTTLNumber(number);
```

Comment:
why set it in a class member again, can't we just set and read it from TablePerperty.properties?

Reply:
analyzePartitionTimeToLive is only responsible for parsing the values in properties, not generating values, so you have to set it